test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -406,8 +427,8 @@ private void performFailoverTest() {
                     txnStatusFutureList.add(checkTxnStatus(transaction, eventWriteCount));
                 } catch (Throwable e) {
                     log.warn("Exception while writing events in the transaction: {}", e);
-                    if (transaction != null) {
-                        log.debug("Transaction with id: {}  failed", transaction.getTxnId());
+                    if (txnDebugReference != null) {
+                        log.debug("Transaction with id: {}  failed", txnDebugReference.getTxnId());
```

Comment:
Q: Should we cancel the the future returned by FutureHelpers.loop incase of an exception? `java.util.concurrent.CompletableFuture#cancel`